subworkflows/local/utils_nfcore_rnaseq_pipeline/main.nf

```diff
@@ -256,6 +256,33 @@ def validateInputParameters() {
         }
     }
 
+    // Check that unaligned reads are saved when using Kraken2
+    if (params.kraken_db) {
+        if (!params.save_unaligned) {
+            error("Please provide --save_unaligned when using Kraken2 to save unaligned reads for analysis.")
+        }
+
+        if (params.aligner == 'star_rsem') {
+            error("Kraken2 cannot be used with --aligner star_rsem since unaligned reads are not saved. Please use --aligner star_salmon or --aligner hisat2.")
```

Comment:
Does RSEM put the unaligned reads into the BAM file? Then they could potentially be extracted from there instead?

Reply:
Based on the [output](https://nf-co.re/rnaseq/3.14.0/docs/output/), there are output BAM files for read alignments to both the genome and the transcriptome. However, RSEM currently does not support `save_unaligned`. It may make sense for there to be a separate PR at another time to add `save_unaligned` to RSEM and also add RSEM to contamination screening. However, I think @Shaun-Regenbaum mentioned there may be other issues with looking at unaligned RSEM reads.